--- a/feeds/packages/net/cups/Makefile
+++ b/feeds/packages/net/cups/Makefile
@@ -30,7 +30,7 @@ define Package/cups
   SECTION:=net
   CATEGORY:=Network
   TITLE:=Common Unix Printing System
-  URL:=http://www.cups.org/
+  URL:=https://www.cups.org/
   DEPENDS:=+libcups +libgnutls +libpthread +libusb-1.0 +zlib
 endef
 
@@ -40,6 +40,13 @@ define Package/cups/description
   web-based administration interface.
 endef
 
+define Package/cups-locale-zh-cn
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=CUPS Chinese (Simplified) language support
+  DEPENDS:=+cups
+endef
+
 define Package/cups-bjnp
   SECTION:=net
   CATEGORY:=Network
@@ -178,6 +185,11 @@ define Package/cups/install
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) ./files/cupsd.init $(1)/etc/init.d/cups
 	$(INSTALL_DIR) $(1)/etc/cups
+	
+	# THDN-PrintServer: 添加中文支持
+	$(INSTALL_DIR) $(1)/usr/share/locale/zh_CN/LC_MESSAGES
+	$(INSTALL_DATA) ./files/cups_zh_CN.mo $(1)/usr/share/locale/zh_CN/LC_MESSAGES/cups.mo
+	
 	$(INSTALL_CONF) ./files/cupsd.conf $(1)/etc/cups/cupsd.conf
 	$(INSTALL_CONF) ./files/client.conf $(1)/etc/cups/client.conf
 	$(INSTALL_DIR) $(1)/usr/lib/cups
@@ -186,6 +198,14 @@ define Package/cups/install
 	$(INSTALL_BIN) ./files/cgi-bin $(1)/usr/lib/cups/cgi-bin
 endef
 
+define Package/cups-locale-zh-cn/install
+	$(INSTALL_DIR) $(1)/usr/share/locale/zh_CN/LC_MESSAGES
+	$(INSTALL_DATA) ./files/cups_zh_CN.mo $(1)/usr/share/locale/zh_CN/LC_MESSAGES/cups.mo
+endef
+
+$(eval $(call BuildPackage,cups-locale-zh-cn))
+
 define Package/cups-bjnp/install
 	$(INSTALL_DIR) $(1)/usr/lib/cups/backend
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/backend/bjnp $(1)/usr/lib/cups/backend
